app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,108 @@
+class OrdersController < ApplicationController
+  before_action :set_orderable
+  before_action :set_order, only: %i(update)
+  before_action :authenticate_user!
+
+  def index
+    if (class_name(@orderable, t("users.user")) && @orderable == current_user) ||
+      (class_name(@orderable, t("parkings.parking")) && @orderable.user == current_user)
+      @orders = @orderable.orders.order_by_time.page(params[:page])
+      if class_name(@orderable, t("parkings.parking"))
+        @orders.each do |order|
+          if deadline_day(order, @orderable) < Time.now && order.status == t("orders.booked")
+            order.update_attribute(:status, t("orders.canceled"))
+            if order.user.score != 1
+              @orderable.update_attribute(:avail_position, @orderable.avail_position + 1)
+              order.user.update_attribute(:score, order.user.score - 3)
+            end
+          end
+        end
+      end
+    else
+      flash[:danger] = t("parkings.not_have_access_rights")
+      redirect_to root_url
+    end
+  end
+
+  def new
+    @order_exist = current_user.orders.find_order(@orderable.id,
+      @orderable.class.name).last
+    if @order_exist.nil? || @order_exist.status == t("orders.sended") ||
```

Comment:
cho sender, canceled cho vào settings theo dạng array nhé rồi check
```suggestion
    if @order_exist.nil? || @order_exist.include? [sended, canceled]
```